src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1217,8 +1217,10 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             return;
         }
 
+        // clamp the converted position to be within the viewport bounds
+        // x: allow range of [-1, RightExclusive].
         til::point terminalPosition{
-            std::clamp(position.x, 0, _terminal->GetViewport().Width() - 1),
+            std::clamp(position.x, -1, _terminal->GetViewport().Width()),

```

Comment:
i see the comment says "allow range of", but... i can see that from the code too. _why_?

Reply:
That's fair. Changed the range to [0, RightExclusive] and it still works fine for shift+click and click+drag scenarios (which seems to be where it's used). Updated to that range and updated the comment.